### PR TITLE
Hit Skipping in Track Candidate producer

### PR DESCRIPTION
**Pull request by a-kapoor**
Hits skipped by seed are also skipped by track.

Number of crossed layers check was also removed.

### CODE REVIEW

**Comment by ANSH0712** [comment]
please test

File: FastSimulation/Tracking/plugins/TrackCandidateProducer.cc [hunk 1]
```diff
@@ -147,9 +147,26 @@ TrackCandidateProducer::produce(edm::Event& e, const edm::EventSetup& es) {
...
     TrajectorySeedHitCandidate recHitCandidate;
-    unsigned numberOfCrossedLayers = 0;      
+    TrajectorySeed::range seedHitRange = seed.recHits();//Hits in a seed
+    for (TrajectorySeed::const_iterator ihit = seedHitRange.first; ihit != seedHitRange.second; ++ihit) {
+      TrajectorySeedHitCandidate seedHit=TrajectorySeedHitCandidate((const FastTrackerRecHit*)(&*ihit),
+								    trackerGeometry.product(),trackerTopology.product());
+      recHitCandidates.push_back(seedHit);
+    }
+    bool passedLastSeedHit = false;
+    const FastTrackerRecHit * lastSeedHit = recHitCandidates.back().hit();
```
Comment (lveldere) [inline FastSimulation/Tracking/plugins/TrackCandidateProducer.cc]: not safe for hitless seeds
and for hitless seeds we need 
```
passedLastSeedHits = true
```

Comment (ANSH0712) [inline FastSimulation/Tracking/plugins/TrackCandidateProducer.cc]: Done !!

**Comment by lveldere** [comment]
please test

**Comment by ANSH0712** [comment]
please test

**Comment by lveldere** [comment]
please test

**Comment by lveldere** [comment]
@ANSH0712 

could you validate this once more, exactly as it is right now?
The comparisons in the github tests look slightly odd, but it might be the limited statistics (only 10 events).

**Comment by ANSH0712** [comment]
@lveldere 

Everything looks fine to me. Here are the validation plots for the latest version:
/afs/cern.ch/user/a/akapoor/public/hitSkipping_9Feb/plots_hitSkipping_9Feb.tgz

**Comment by lveldere** [comment]
+1

thanks @ANSH0712

**Comment by davidlange6** [comment]
+1

**Pull request merged**